src/main/java/de/tum/in/www1/artemis/service/connectors/localci/buildagent/LocalCIDockerService.java

```diff
@@ -290,4 +299,31 @@ public void deleteOldDockerImages() {
             }
         }
     }
+
+    /**
+     * Periodically checks for available disk space and triggers the cleanup of old Docker images if the available space falls below a specified threshold.
+     * The threshold is configurable via the 'artemis.continuous-integration.image-cleanup.disk-space-threshold-mb' property in the application settings.
+     */
+
+    @Scheduled(fixedRateString = "${artemis.continuous-integration.image-cleanup.disk-space-check-interval-ms:3600000}")
+    public void CheckUsableDiskSpaceThenCleanUp() {
+        if (imageCleanupEnabled) {
+            try {
+                // Get the root partition.
+                File file = new File(Objects.requireNonNullElse(dockerClient.infoCmd().exec().getDockerRootDir(), "/"));
+                // Get the usable space in bytes.
+                long usableSpace = file.getUsableSpace();
+                // Get the threshold in bytes.
+                long threshold = imageCleanupDiskSpaceThresholdMb * 1024 * 1024L;
```

Comment:
I think it would be good to get rid of the magic numbers, could you use descriptive variable names for them instead?